server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java

```diff
@@ -525,7 +531,13 @@ public void apply(Settings value, Settings current, Settings previous) {
         MasterHistory.MAX_HISTORY_AGE_SETTING,
         ReadinessService.PORT,
         HealthNode.isEnabled() ? HealthNodeTaskExecutor.ENABLED_SETTING : null
-    ).filter(Objects::nonNull).collect(Collectors.toSet());
+    )
+        .filter(Objects::nonNull)
+        .filter(
```

Comment:
I think I would prefer to check the feature flag inline with the setting, like line 533 above. I'll suggest what I mean on the line above, but that would mean we'd revert this hunk.

Reply:
I agree. I made the change.